### ISSUE TITLE
Symbol keys of properties are indistinguishable from string keys

### ISSUE THREAD

**Issue by ajvincent** [typescript-eslint/typescript-eslint#5226]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&showAST=es&code=MYewdgzgLgBAHjAvDAygTwLYCMQBsAUARHIQJQDcAUKJLAA5IwDelMMaAXKpjgYVGUoBfKpQBmAVzDAoAS3Awo+Us1YwATgFMoE9WBjFCVIZWrhoMAF6MlFUzQtpGLNnC5iAhrgiaANGoBtOABdLih1CT9Ay1DFCKiTIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const x = Symbol("x");
const p = {
  y: Symbol("t")
};

function t() {
  return "x";
}

const z = t();

const y = {
  x: false,
  [x]: true,
  [z]: true,
}
```

### Expected Result

The three Property nodes should have some unique, observable difference other than the `range` and `loc` properties.

### Actual Result

For the variable declaration `y`:
- The first property node has key: Identfier with name x, and computed: false
- The second property node has key: Identifier with name x, and computed: true
- The third property node has key: Identifier with name z, and computed: true
 
When parsing the two `x` property identifiers of `y` in my example, they have the same type and name.  The `Property` nodes look identical _except_ for the `range` and `loc` properties, which is to be expected.

In particular, there's really no way to tell from the specific Property node that one is a string key and the other is a [symbol key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol).  The third property node illustrates that the `computed` attribute of the `Property` type is not sufficient.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `5.29.0` |
| `TypeScript`                           | `4.7.3` |
| `ESLint`                               | `8.17.0` |
| `node`                                 | `16.14.0` |

**Comment by ajvincent** [comment]
Upon further reflection, this bug may _technically_ be invalid.  The information to resolve this is available elsewhere in the tree, but it requires knowledge of scopes and looking up where the key came from.  (Scope manager?)

**Comment by bradzacher** [comment]
The AST represents the code as written.
As written it doesn't matter the value of the variable of a computed property - just that its a variable.

It does suck that the only AST difference between a "named property" and a "computed property with a calculated name that is a variable" is the `computed` flag - but that's just the AST we have to work with - it's not our design - it's the ESTree AST representation.
It is impossible to make breaking changes to the AST spec, sadly.

If you want to determine the type of a variable - you can use type information to do so.
Otherwise you can leverage scope analysis to do so. Though be aware scope analysis does not work across files.
